### PR TITLE
Add Timing Closure Guide

### LINKED ISSUE #412: Integrate @shalan's Timing Closure Documentation
### Description

@shalan made an excellent document on timing violations and achieving timing closure for OpenLane. https://docs.google.com/document/d/13J1AY1zhzxur8vaFs3rRW9ZWX113rSDs63LezOOoXZ8/edit

Unfortunately, it is a Google Doc and OpenLane variable names have changed since then. 

### Proposal

Integrate this into the OpenLane documentation proper.

### PR DESCRIPTION
**Pull request by donn**
## Docs
* Adapted timing closure guide by @shalan to OpenLane 2
  * Converted to MyST Markdown
  * All images made dark-mode friendly
  * References to variables all now resolve properly
* Fixed a number of warnings and broken links in the documentation.
* Subfigures and citations now supported.

## Tool Updates
* `mdformat` patches now mess up MyST files less (`mdformat-myst` doesn't work)

---
Resolves #412

### CODE REVIEW

File: docs/source/usage/caravel/index.md [hunk 1]
```diff
@@ -1516,7 +1516,7 @@ There are around 260 antenna violations with ratios up to 7.
 ⋮
 ```
 
-We can fix those the same way we did in the AES [here](openroad-checkantennas).
+We can fix those the same way we did in the AES [here](#openroadcheckantennas).
```
Comment (kareefardi) [inline docs/source/usage/caravel/index.md:1519]: Not the right reference.

File: docs/source/usage/caravel/index.md [hunk 2]
```diff
@@ -1544,7 +1544,7 @@ violations and 1 hold violation which is not Reg to Reg.
 ```
 
 The max Slew/Cap violations can be fixed the same way in
-[openroad-stapostpnr](openroad-stapostpnr). For the hold violation, it is in the
+[openroad-stapostpnr](#openroadstapostpnr). For the hold violation, it is in the
```
Comment (kareefardi) [inline docs/source/usage/caravel/index.md:1547]: Not the right reference

**Review by kareefardi — CHANGES_REQUESTED** [review]
There are a couple of warnings/errors generated by myst. Please address them.

File: docs/source/usage/caravel/index.md (hunk 1 above)
Comment (donn) [reply docs/source/usage/caravel/index.md:1519]: I tried it, it worked, can you please confirm this is broken?

File: docs/source/usage/caravel/index.md (hunk 2 above)
Comment (donn) [reply docs/source/usage/caravel/index.md:1547]: I tried it, it worked, can you please confirm this is broken?

File: docs/source/usage/caravel/index.md (hunk 1 above)
Comment (kareefardi) [reply docs/source/usage/caravel/index.md:1519]: It is not broken. It is jumping to a different section than the one in master.

Comment (donn) [reply docs/source/usage/caravel/index.md]: I added explicit section links for the intended sections.

**Review by kareefardi — APPROVED** [review]

**Review by kareefardi — APPROVED** [review]

**Pull request merged**